lib/executables/validate.js

```diff
@@ -0,0 +1,146 @@
+        const viewClass = assert.error(fields[field].viewClass, `field '${field}' does not have a view class`);
+        if (viewClass) {
+          assert.warn(VIEW_CLASSES.includes(fields[field].viewClass), `view class '${fields[field].viewClass}' in credential field '${field}' is not valid
+            => Valid username classes are [${VIEW_CLASSES}]`);
+        }
+      });
+    }
+  }
+
+  print.info('Checking actions...');
+  assert.warn(component.actions, 'No actions have been found');
+
+  if (component.actions) {
+    Object.keys(component.actions).forEach((action) => {
+      const details = component.actions[action];
+      checkDetails(details, action, pathName, 'action');
+    });
+  }
+
+  if (valid) assert.info('All actions were validated!');
+  valid = false;
+
+  print.info('Checking triggers...');
+  assert.warn(component.triggers, 'No triggers have been found');
+
+  if (component.triggers) {
+    Object.keys(component.triggers).forEach((trigger) => {
+      const details = component.triggers[trigger];
+      checkDetails(details, trigger, pathName, 'trigger');
+    });
+  }
+
+  if (valid) assert.info('All actions were validated!');
+  valid = false;
+
+  print.info('Checking schema files...');
+
+  Object.keys(schemas).forEach((name) => {
+    const schema = schemas[name];
+    let file; let schemaExists;
+
+    if (typeof schema === 'string') {
+      schemaExists = fs.existsSync(path.resolve(pathName, schema));
+      assert.error(schemaExists, `Schema ${name} located at ${schema} does not exist`);
+      if (schemaExists) {
+        // eslint-disable-next-line import/no-dynamic-require, global-require
+        file = require(path.resolve(pathName, schema));
+      }
+    }
+    // eslint-disable-next-line no-unused-expressions
+    file ? JSON.stringify(file) : JSON.stringify(schema);
+  });
+
+  if (valid) assert.info('All schema files are valid!');
+  else if (!Object.keys(schemas).length) assert.info('No static schema files found');
+};
+
+const checkDetails = (details, name, pathName, type) => {
+  assert.error(details.title, `${type} '${name}' is missing a title`);
+  assert.warn(details.description, `${type} '${name}' is missing a description`);
+  assert.error(details.main, `no file defined for ${type} '${name}'`);
+  const actionExists = fs.existsSync(path.resolve(pathName, details.main));
+  assert.error(actionExists, `file pathname for ${type} '${name}' does not lead to a file`);
+
+  if (actionExists) {
+    try {
+      // eslint-disable-next-line import/no-dynamic-require, global-require
+      const actionFile = require(path.resolve(pathName, details.main));
+      assert.error(actionFile.process, `${type} '${name}' does not have a process function`);
+      if (details.dynamicMetadata) {
+        assert.error(actionFile.getMetaModel, `${type} '${name}' has dynamic metadata specified but no getMetaModel function`);
+      }
```

Comment:
* There is another level of checks to ensure that any dynamic dropdowns have matching functionality.
* There is another layer of checks about `init()`, `startup()` and `shutdown()`.
* It makes sense to check that `process`  is a function (`typeof process === 'function'`)  and if [it is async](https://davidwalsh.name/javascript-detect-async-function).